tests/test_wasapi_client.py

```diff
@@ -628,10 +636,13 @@ def test_run_WASAPIDownloadError(self, mock_download):
         p.run()
         # If the join doesn't block, the queue is fully processed.
         get_q.join()
-        assert result_q.qsize() == 2
-        assert log_q.qsize() == 2
         for _ in (1, 2):
+            assert log_q.get().msg == expected_error
             assert result_q.get() == ('failure', self.filename)
+        # Verify those were the only two results on the result_q.
+        # Sometimes `empty` needs a moment to register.
+        time.sleep(.5)
```

Comment:
I think the changes here generally look good, but with this one, while I understand `empty()` may need a second to "catch up", why do we _only_ use sleep before it _here_, and not other places as well?

Reply:
Essentially I'm wondering if we're randomly going to encounter errors in other tests due to this same issue.

Reply:
I had some other sleeps in there, but then found it didn't seem to be needed in the other places (in checking different sleep times) and took them out. So far, this is the only one that has caused a problem...

Reply:
Switched to manager.Queue! seems like we use in wasapi_client.py. Seems to take care of the issue and avoid `sleep`.

Reply:
Yeah, I think that's the best way to take care of this. :thumbsup: